_posts/2017-08-18-how-to-check-the-spelling-of-your-docs-from-travis-ci.md

```diff
@@ -0,0 +1,133 @@
+
+Note that you might want to execute this command many times, in English and French for example, especially if you write technical documents in French that will also contain many English words.
+
+Of course you will also need to allow personal words. To do so, you can add and use custom dictionaries files, named `.aspell.en.pws` for an English personal dictionary. Here is an example of what this file could contain:
+
+```
+personal_ws-1.1 en 3
+php
+javascript
+android
+```
+
+Note that the header (first line) of this file is important: the two last arguments correspond to the language and the number of words.
+
+Then to use this personal dictionary, you have to add this argument in your command: `--personal=./.aspell.en.pws`
+
+How to execute this tool from Travis CI?
+========================================
+
+To execute `aspell` from Travis CI, you need to install this apt package in the Travis container. To do so, add these lines in your `.travis.yml` file:
+
+```yml
+addons:
+  apt:
+    packages:
+      - aspell
+      - aspell-en
+      - aspell-fr # that corresponds to your language
+```
+
+Then, in this same configuration file, you can execute a custom bash script:
+
+```yml
+script: your_script.sh
+```
+
+In this script, you can use the environment variable `$TRAVIS_COMMIT_RANGE`, available in the Travis container, to get only the files that were changed in the pull request that corresponds to your build:
+
+```bash
+git diff --name-only $TRAVIS_COMMIT_RANGE
+```
+
+If you want to get only markdown files, you can add `| grep .md` at the end of the previous command.
+
+Once you have the names of the files that should be checked for this pull request, you can execute the `aspell list` command that we've seen in the first part.
+
+Note that you can also use `grep` and `sed` commands to remove metadata or code blocks from your files before executing `aspell` command, if you don't want to check the spelling in these blocks.
+For example, if you want to remove your code blocks from your markdown file, you can use this command:
+
+```bash
+cat your_file.md | sed  -n '/^```/,/^```/ !p'
+```
+
+How to send the results to Github pull request?
+===============================================
+
+We don't want this script to block the reviewers to merge the pull request, so the first thing to do is to add `exit 0` at the end of the script that will be executed from Travis CI. Otherwise if an error code is returned by the script, Travis will mark the pull request status as failing, and will block the user from merging this merge request.
+
+The easiest thing that we can do to send the results of previous commands is to post them in a comment on the Github pull request.
+
+First you need to choose the Github user that will be used to add this comment, and configure an access token for this user:
+- Login to [https://github.com](https://github.com) with this user
+- Go to [https://github.com/settings/tokens](https://github.com/settings/tokens)
```

Comment:
I think you should give this link here: `https://github.com/settings/tokens/new` (because you don't ask to click on "Generate new token" button and it's a kind of shortcut)